src/Sulu/Component/Content/Tests/Functional/Repository/ContentRepositoryTest.php

```diff
@@ -719,7 +720,84 @@ public function testFindPermissions()
         );
 
         $this->assertEquals(
-            [1 => ['edit' => true], 2 => ['view' => true, 'archive' => true]],
+            [
+                1 => ['view' => false, 'add' => false, 'delete' => false, 'edit' => true],
+                2 => ['view' => true, 'add' => false, 'edit' => false, 'delete' => false, 'archive' => true]
+            ],
+            $result->getPermissions()
+        );
+    }
+
+    public function testFindWithoutPermissions()
+    {
+        $role1 = $this->prophesize(RoleInterface::class);
+        $role1->getId()->willReturn(1);
+        $role1->getIdentifier()->willReturn('ROLE_SULU_ROLE 1');
+        $role2 = $this->prophesize(RoleInterface::class);
+        $role2->getId()->willReturn(2);
+        $role2->getIdentifier()->willReturn('ROLE_SULU_ROLE-2');
+
+        $user = $this->prophesize(UserInterface::class);
+        $user->getRoleObjects()->willReturn([$role1->reveal(), $role2->reveal()]);
+
+        $page = $this->createPage(
+            'test-1',
+            'de',
+            [],
+            null
+        );
+
+        $result = $this->contentRepository->find(
+            $page->getUuid(),
+            'de',
+            'sulu_io',
+            MappingBuilder::create()->getMapping(),
+            $user->reveal()
+        );
+
+        $this->assertEquals(
+            [],
+            $result->getPermissions()
+        );
+    }
+
+    public function testFindWithPermissionsNotGranted()
+    {
+        $role1 = $this->prophesize(RoleInterface::class);
+        $role1->getId()->willReturn(1);
+        $role1->getIdentifier()->willReturn('ROLE_SULU_ROLE 1');
+        $role2 = $this->prophesize(RoleInterface::class);
+        $role2->getId()->willReturn(2);
+        $role2->getIdentifier()->willReturn('ROLE_SULU_ROLE-2');
+
+        $user = $this->prophesize(UserInterface::class);
+        $user->getRoleObjects()->willReturn([$role1->reveal(), $role2->reveal()]);
+
+        $page = $this->createPage(
+            'test-1',
+            'de',
+            [],
+            null,
+            [
+                1 => ['edit' => false],
+                2 => ['view' => false, 'archive' => false],
+                3 => ['add' => false],
```

Comment:
This results in an error with a jackalope version <1.3. Shall we raise the minimum dependency, or should we just skip this test if that version is installed? (would require reading the `composer.lock` file)